plugins/common/shim/config.go

```diff
@@ -0,0 +1,168 @@
+package shim
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/BurntSushi/toml"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+type config struct {
+	Inputs     map[string][]toml.Primitive
+	Processors map[string][]toml.Primitive
+	Outputs    map[string][]toml.Primitive
+}
+
+type loadedConfig struct {
+	Input     telegraf.Input
+	Processor telegraf.StreamingProcessor
+	Output    telegraf.Output
+}
+
+// LoadConfig Adds plugins to the shim
+func (s *Shim) LoadConfig(filePath *string) error {
+	conf, err := LoadConfig(filePath)
+	if err != nil {
+		return err
+	}
+	if conf.Input != nil {
+		if err = s.AddInput(conf.Input); err != nil {
+			return fmt.Errorf("Failed to add Input: %w", err)
+		}
+	} else if conf.Processor != nil {
+		if err = s.AddStreamingProcessor(conf.Processor); err != nil {
+			return fmt.Errorf("Failed to add Processor: %w", err)
+		}
+	} else if conf.Output != nil {
+		if err = s.AddOutput(conf.Output); err != nil {
```

Comment:
Does this let people have multiple inputs, processors, and outputs in the same shim?  I guess I expected one plugin of any type per shim process.

Reply:
yeah, you can only have one.